core/src/main/resources/error/error-classes.json

```diff
@@ -36,6 +73,18 @@
   "INVALID_JSON_SCHEMA_MAPTYPE" : {
     "message" : [ "Input schema %s can only contain StringType as a key type for a MapType." ]
   },
+  "INVALID_PARTITION_FILTER_ERROR" : {
+    "message" : [ "Partition filter cannot have both `\"` and `\\'` characters" ],
+    "sqlState" : "42000"
+  },
+  "LEGACY_METADATA_PATH_EXISTS_ERROR" : {
+    "message" : [ "Error: we detected a possible problem with the location of your \"_spark_metadata\"\ndirectory and you likely need to move it before restarting this query.\n\nEarlier version of Spark incorrectly escaped paths when writing out the\n\"_spark_metadata\" directory for structured streaming. While this was corrected in\nSpark 3.0, it appears that your query was started using an earlier version that\nincorrectly handled the \"_spark_metadata\" path.\n\nCorrect \"_spark_metadata\" Directory: %s\nIncorrect \"_spark_metadata\" Directory: %s\n\nPlease move the data from the incorrect directory to the correct one, delete the\nincorrect directory, and then restart this query. If you believe you are receiving\nthis message in error, you can disable it with the SQL conf\n%s." ],
```

Comment:
You can split the message here into array elements, which will be re-joined with newlines later.